### DISCUSSION TITLE
Copy data - advise for approach with ormar

### DISCUSSION THREAD

**Discussion by avico78** [collerek/ormar#177] [Q&A]
I'm checking the option for using ormar as copy data from source db to target db_b .

Assumptions :
tables has similar name and almost identical structure where it possible that table on target can have slight changes:
   Extra fields
   Missing fields 
   changes in field length 

I'm trying to come up with approach achieving this and also solution that can be flexible and be maintainable in reasonable efforts.

**Comment by collerek** [comment]
And what is the volume of data and number of the tables? You want exact copies of all columns including primary keys?

**Reply by avico78** [reply]
@collerek 

Thanks ,
the volume  of data will be decide according the solution capabilities ,
hopefully it can provide high performance .

regearding the 
for PK , Uniq constrains ,index  - can be painfull when copy to target DB which has already existing data ,
so currently I have assumption that there will not be collision/violation whily copy from sourc dB to target DB , maybe later i will think about some pre-copy process
"mark" data on source which has collisions which target and exclude it from the copy list .
but surely will need to handle that (pk,index).
just reminder ,the data copy is on distributed env
db_source_a --> db_target_a
db_source_b --> db_target_b

as we discuss previously ,data commit only if no error occurred in all parallel session .
I open minded and open for any suggestion for the approach (the ormar approach :))
